lesson07/assignment/assignment07/src/App.vue

```diff
@@ -0,0 +1,85 @@
+<template>
+  <div id="app">
+    <a href="#main-content" class="hidden">Skip to main content</a>
+    <div class="container-fluid">
+      <div id="nav">
+        <div class="row">
+           <!-- JUMBOTRON AND NAV -->
+          <header class="col-sm-12 col-12 jumbotron jumbotron-fluid background-image">
+            <nav class="navbar navbar-expand-lg d-flex flex-row justify-content-between align-items-start">
+              <div class="navbar-brand text-white text-uppercase d-flex flex-column align-items-center">
+                <h1 class="display-1">Ten Ton</h1>
+                <h1 class="display-3">Coffee</h1>
+              </div>
+              <ul class="navbar-nav text-uppercase">
+                <li class="nav-item"><router-link class="text-white nav-link" to="/">Home</router-link></li> 
+                <li class="nav-item"><router-link class="text-white nav-link" to="/about">About</router-link></li> 
+                <li class="nav-item"><router-link class="text-white nav-link" to="/menu">Menu</router-link></li> 
+                <li class="nav-item"><router-link class="text-white nav-link" to="/contact">Contact</router-link></li>
+              </ul>
+            </nav>
+          </header>
+        </div>
+        <router-view/>
+      </div>
+
+      <!-- FOOTER -->
+        <footer class="text-white pt-4 pb-1 px-4">
+          <div class="row">
+            <form class="form-col col-md-7 col-lg-6">
+              <legend>
+                <h4>Find out what's brewing at Ten Ton Coffee</h4>
+                <p>Sign up for our mailing list and receive 15% off your next order</p>
+              </legend>
+              <!-- TOOLTIPS -->
+              <div class="form-inline">
+                <input id="input" class="w-75 form-control form-control-lg rounded-sm border border-0" type="email" name="email" placeholder="Enter your email" aria-label="Email" data-toggle="tooltip" data-placement="bottom" title="Enter email to subscribe!">
+                <!-- ONCLICK AND IF/ELSE -->
+                <button v-on:click="isEmpty" class="subscribe text-uppercase rounded-sm py-2 px-2" type="submit" data-toggle="tooltip" data-placement="bottom" title="Subscribe here!">Subscribe</button>
+              </div>
+              <div id="isEmptyText" v-if="isEmpty"></div>
```

Comment:
Excellent work on validation and user messaging! Consider switching naming here to "isValid"/"isValidText" rather than "isEmpty", as it's more accurate.